Implement a `hasPseudoElementStyle` assertion

joankaradimov opened the pull request:
Resolves https://github.com/simplabs/qunit-dom/issues/229

Turbo87 commented on lib/assertions.ts:
```diff
@@ -492,15 +492,41 @@ export default class DOMAssertions {
+   *
+   * @see {@link #hasClass}
+   */
+  hasPseudoElementStyle(selector: string, expected: object, message?: string): void {
```
I **think** this would have to be `selector: string | null`, and then we'd need another variant above that accepts `selector: string` as the public interface to that method. If I remember correctly the actual implementation is always hidden if you have specializations like the `selector: null` above 🤔

joankaradimov replied:
Will do.

joankaradimov commented:
> nice work @joankaradimov!

Thanks!

> just one minor thing left to do :)

I've committed the suggested change.

Just let me know if everything else looks good so that I squash the commits.

Turbo87 commented on lib/assertions.ts:
```diff
@@ -492,15 +492,42 @@ export default class DOMAssertions {
+   * });
+   *
+   * @see {@link #hasClass}
+   */
```
can you check that `yarn docs` reads the comment correctly? the `API.md` change should then also be committed, I guess

joankaradimov replied:
It does not work correctly.

joankaradimov commented:
It turns out the initial overload signature for `hasPseudoElementStyle` was the correct one. The one in a858a7386012f7aa7779fc47cc84f5ea3aa997f9 does not generate any documentation. I've reverted it.

Also there seemed to be some code which did not have a documentation generated for it. I've committed that separately.

Finally - there are some DOS new lines mixed with the Unix new lines in the examples in API.md. This appears to be the case for the existing examples too. There are no DOS new lines in the docstrings. Should I log a bug for that?

Turbo87 commented:
> there are some DOS new lines mixed with the Unix new lines in the examples in API.md. This appears to be the case for the existing examples too. There are no DOS new lines in the docstrings. Should I log a bug for that?

Interesting. That sounds like a bug (or maybe even feature 🤔) in `documentation` to me

Turbo87 commented:
@joankaradimov the problem with `documentation` is that is does not pick up the docs from these method specializations. I've moved the docstring on the method implementation, while keeping the `selector: string` specialization and now it seems to work as intended :)

joankaradimov commented:
> I've moved the docstring on the method implementation

Ah, ok. I was thinking of something different.

> there seemed to be some code which did not have a documentation generated for it.

I see you made this private and removed the commit.

I'm going to squash the rest of it and force push.

Turbo87 commented:
thanks again 🙏

joankaradimov commented:
Thank you for the help and the quick responses. And also for quickly releasing this!
